WikiWatershed/model-my-watershed
# Add TDX Streamlines Subset Visual Layer

## Overview

This PR adds a demo layer for a TDX Streamlines subset for the US East Coast. The data was pulled from NGA and ingested into vector tiles, as detailed in #3623. These were uploaded to the `tiles.us-east-1.azavea.com` bucket in the `azavea-datahub` account, which also host other static tiles used in MMW.

Since we use an old version of Leaflet in this project, we had to use [Leaflet.MapboxVectorTile](https://github.com/SpatialServer/Leaflet.MapboxVectorTile) instead of [Leaflet.VectorGrid](https://github.com/Leaflet/Leaflet.VectorGrid). Furthermore, I had to [fork the project](https://github.com/rajadain/Leaflet.MapboxVectorTile) to [make it compatible with our import method](https://github.com/rajadain/Leaflet.MapboxVectorTile/commit/842921d6b93cc1d78960aabbdb13a96ab63b0bc3). The resulting file has now been added directly as a shim, so we don't need to maintain my fork going forward. It is there only for historical purposes.

The layer added here is just for demo purposes. We need to improve styling, stitching, and overzooming before this can hit production.

Closes #3623 

### Demo

https://github.com/WikiWatershed/model-my-watershed/assets/1430060/5f8b5b9e-60ae-4324-aa76-851bdf501066

## Testing Instructions

* Check out this branch
* Run `./scripts/bundle.sh --vendor`
* Go to http://localhost:8000/
* Pick the TDX streams in the layer selector
  - [ ] Ensure you see them load correctly
  - [ ] Ensure they are performant and load quickly
